### DISCUSSION TITLE
Using Energyplus on linux and Mac

### DISCUSSION THREAD

**Discussion by zha** [NREL/EnergyPlus#8609] [Q&A] [answered]
I noticed that from version 9.3 and onwards, I am no longer able to use the pre-compiled Energyplus packages on Linux and Mac. It just gave me the following error when I run the simulation:
```
** Severe  ** Failed to import module "pluginA"
**   ~~~   ** Python error description follows: 
**   ~~~   ** ImportError('libffi.so.6: cannot open shared object file: No such file or directory',)
**  Fatal  ** Python import error causes program termination
```

I am pretty sure that this is due to the recent addition of Python EMS module.  To get around this, I always have to download the source code and use CMake to build the EP. I am certainly OK with taking extra steps, but I am just wondering if this is something that will be fixed soon?

Thanks.

**Comment by Myoldmopar** [comment, marked as answer]
We routinely develop, test, and package EnergyPlus on Ubuntu 18.04 and 20.04, and Mac OSX, so we should be able to get you going.   I know you said "Linux and Mac", but that is too broad.  For Linux we have different packages for Ubuntu 18.04 and 20.04.  If you are using a different version of Ubuntu or a different distribution of Linux, we won't have as much control over the dependencies available, but we can still try to get you going.  And what version of OSX are you running?

**Reply by zha** [reply]
Thank you for your reply.

I was running EnergyPlus on a HPC cluster. It runs CentOS 7. On Mac, I was using MacOS 10.15.7.

Thank you.

**Reply by Myoldmopar** [reply]
We don't officially test or release for CentOS, but I bet we can get it running.  The error above says it is missing the ffi library.  I think the best way to fix it would be to just `rpm install` whatever the ffi library is called.  However, I'm guessing you don't have auth to do that :upside_down_face: on your cluster.  There are a couple alternatives.  
1. You could try to build EnergyPlus right on the cluster and then you'd have a package that is ready to run with that exact system dependencies.  
2. I could try to build you a package on Ubuntu, but with Python support turned off.  You wouldn't be able to run any Python Plugin files, but you wouldn't have this exact same dependency issue.  I can't say that it wouldn't lead to simply a different dependency.

As for Mac, our CI system builds the Mac packages on 10.15.7, so I am quite surprised that you are having an issue there.  Is it the same error message?

**Reply by zha** [reply]
Thanks @Myoldmopar .

That's correct. I cannot run `rpm` on the cluster; as I said in the OP, I had to build EnergyPlus to get it working. 

TBH, for MacOS I think that error might have come from pre-built EP 9.3. I haven't tested it out on the newer EP 9.4 (pre-built) just yet as I have been only using Linux on the HPC for now. However, after I built the EP on MacOS just like I did on Linux, then everything was just fine.
